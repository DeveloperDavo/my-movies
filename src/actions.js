export const changeMinRating = minRating => {
  return {
    type: 'changeMinRating',
    minRating
  }
}
