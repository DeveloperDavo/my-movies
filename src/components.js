import React, { Component } from "react";
import styled from 'styled-components';

const smallScreen = '(min-width: 576px)'
const largeScreen = '(min-width: 992px)'

export const Header = () => {
  const Title = styled.h1`
    margin: 0;
    font-size: 3rem;
  `;

  const HeaderWrapper = styled.header`
    background-color: #222;
    color: white;
  `;

  return (
    <HeaderWrapper>
      <Title>My Movies</Title>
    </HeaderWrapper>
  );
};

const MinRatingSelect = ({ value, onChange }) => {
  const Select = styled.select`
    display: inline-block;
    height: 2rem;
    vertical-align: middle;
    border: 1px solid #ced4da;
  `;

  const options = [<option key="0" value="0">none</option>];
  for (let i = 5; i <= 10; i = i + 0.5) {
    options.push(<option key={i} value={i}>{i}</option>);
  }
  return (
    <div id="MinRatingSelect">
      <label>Choose a minimum rating: </label>
      <Select value={value} onChange={onChange}>
        {options}
      </Select>
    </div>
  );
};

const MovieGridItem = ({ movie, imageSrc }) => {
  const MovieGridItem = styled.div`
    margin: 1rem;
    padding: 1rem;
    background-color: white;
  `;

  return (
    <MovieGridItem name='MovieGridItem'>
      <img
        width="200"
        src={imageSrc}
        alt={movie.title}
      />
      <h3>{movie.title}</h3>
      <div>Rating: {movie.vote_average}</div>
      <div>Review Count: {movie.vote_count}</div>
    </MovieGridItem>
  )

}

export const MovieGrid = ({ movies, minRating }) => {
  const MovieGrid = styled.div`
    display: grid;
    grid-template-columns: 1fr;
    @media ${smallScreen} { 
        grid-template-columns: 1fr 1fr;
    @media ${largeScreen} { 
        grid-template-columns: 1fr 1fr 1fr;
    }
  `;

  const movieGrid = movies
    .filter(movie => movie.vote_average >= minRating)
    .map(movie => {
      const imageSrc = `http://image.tmdb.org/t/p/w200/${movie.poster_path}`;
      return (
        <MovieGridItem movie={movie} imageSrc={imageSrc} name='MovieGridItem' key={movie.id} />
      );
    });
  return <MovieGrid>{movieGrid}</MovieGrid>;
};

const Attribution = () => {
  const Footer = styled.footer`
    margin: 1rem;
    display: flex;
    justify-content: center;
  `;
  const Logo = styled.img`
    align-self: center;
    height: 2rem;
  `;
  const AttributionText = styled.div`
    padding: 1rem;
    text-align: initial;
  `;
  return (
    <Footer>
      <Logo src="tmdb-logo.png" alt="tmdb logo"/>
      <AttributionText>
        This product uses the TMDb API but is not endorsed or certified by TMDb.
      </AttributionText>
    </Footer>
  )
}

export class MainComponent extends Component {
  constructor(props) {
    super(props)

    const { handleMinRatingChange, handleFetchMovies } = this.props;
    handleMinRatingChange.bind(this);
    handleFetchMovies.bind(this);
  }


  componentDidMount() {
    this.props.handleFetchMovies()
  }

  render() {
    const Title = styled.div`
      margin: 2rem;
      font-size: 1.5em;
    `;

    const { minRating, handleMinRatingChange, movies } = this.props;
    return (
      <main>
        <Title>Movies now playing</Title>
        <MinRatingSelect
          value={minRating}
          onChange={handleMinRatingChange}
        />
        <MovieGrid movies={movies} minRating={minRating} />
        <Attribution />
      </main>
    );
  }
}