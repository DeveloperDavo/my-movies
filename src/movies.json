{
  "results": [
    {
      "vote_count": 1097,
      "id": 337167,
      "video": false,
      "vote_average": 6.1,
      "title": "Fifty Shades Freed",
      "popularity": 572.203541,
      "poster_path": "/jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
      "original_language": "en",
      "original_title": "Fifty Shades Freed",
      "genre_ids": [18, 10749],
      "backdrop_path": "/9ywA15OAiwjSTvg3cBs9B7kOCBF.jpg",
      "adult": false,
      "overview":
        "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
      "release_date": "2018-02-07"
    },
    {
      "vote_count": 3477,
      "id": 284054,
      "video": false,
      "vote_average": 7.4,
      "title": "Black Panther",
      "popularity": 329.754985,
      "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "original_language": "en",
      "original_title": "Black Panther",
      "genre_ids": [28, 12, 14, 878],
      "backdrop_path": "/b6ZJZHUdMEFECvGiDpJjlfUWela.jpg",
      "adult": false,
      "overview":
        "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \"special forces\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "release_date": "2018-02-13"
    },
    {
      "vote_count": 3404,
      "id": 354912,
      "video": false,
      "vote_average": 7.8,
      "title": "Coco",
      "popularity": 211.906826,
      "poster_path": "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
      "original_language": "en",
      "original_title": "Coco",
      "genre_ids": [12, 35, 10751, 16],
      "backdrop_path": "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
      "adult": false,
      "overview":
        "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
      "release_date": "2017-10-27"
    },
    {
      "vote_count": 588,
      "id": 338970,
      "video": false,
      "vote_average": 6.1,
      "title": "Tomb Raider",
      "popularity": 197.571243,
      "poster_path": "/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
      "original_language": "en",
      "original_title": "Tomb Raider",
      "genre_ids": [28, 12],
      "backdrop_path": "/bLJTjfbZ1c5zSNiAvGYs1Uc82ir.jpg",
      "adult": false,
      "overview":
        "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
      "release_date": "2018-03-08"
    },
    {
      "vote_count": 9,
      "id": 499772,
      "video": false,
      "vote_average": 6,
      "title": "Meet Me In St. Gallen",
      "popularity": 177.406978,
      "poster_path": "/uGntNjUx6YAzbVy7RDgxWnWsdet.jpg",
      "original_language": "tl",
      "original_title": "Meet Me In St. Gallen",
      "genre_ids": [18, 10749],
      "backdrop_path": "/4QrbczSQGZQA7BG9xMhccQI7LHm.jpg",
      "adult": false,
      "overview":
        "The story of Jesse and Celeste who meets at an unexpected time in their lives. They then realize their names are the same as the characters in the popular break-up romantic comedy, Celeste and Jesse Forever.",
      "release_date": "2018-02-07"
    },
    {
      "vote_count": 1023,
      "id": 300668,
      "video": false,
      "vote_average": 6.4,
      "title": "Annihilation",
      "popularity": 136.399701,
      "poster_path": "/d3qcpfNwbAMCNqWDHzPQsUYiUgS.jpg",
      "original_language": "en",
      "original_title": "Annihilation",
      "genre_ids": [9648, 878, 18],
      "backdrop_path": "/5zfVNTrkhMu673zma6qhFzG01ig.jpg",
      "adult": false,
      "overview":
        "A biologist signs up for a dangerous, secret expedition into a mysterious zone where the laws of nature don't apply.",
      "release_date": "2018-02-22"
    },
    {
      "vote_count": 2963,
      "id": 399055,
      "video": false,
      "vote_average": 7.3,
      "title": "The Shape of Water",
      "popularity": 128.39434,
      "poster_path": "/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg",
      "original_language": "en",
      "original_title": "The Shape of Water",
      "genre_ids": [18, 14, 10749],
      "backdrop_path": "/rgyhSn3mINvkuy9iswZK0VLqQO3.jpg",
      "adult": false,
      "overview":
        "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
      "release_date": "2017-12-01"
    },
    {
      "vote_count": 541,
      "id": 460793,
      "video": false,
      "vote_average": 5.9,
      "title": "Olaf's Frozen Adventure",
      "popularity": 108.532582,
      "poster_path": "/As8WTtxXs9e3cBit3ztTf7zoRmm.jpg",
      "original_language": "en",
      "original_title": "Olaf's Frozen Adventure",
      "genre_ids": [12, 16, 35, 10751, 14, 10402],
      "backdrop_path": "/9K4QqQZg4TVXcxBGDiVY4Aey3Rn.jpg",
      "adult": false,
      "overview":
        "Olaf is on a mission to harness the best holiday traditions for Anna, Elsa, and Kristoff.",
      "release_date": "2017-10-27"
    },
    {
      "vote_count": 159,
      "id": 268896,
      "video": false,
      "vote_average": 6.3,
      "title": "Pacific Rim: Uprising",
      "popularity": 89.831077,
      "poster_path": "/v5HlmJK9bdeHxN2QhaFP1ivjX3U.jpg",
      "original_language": "en",
      "original_title": "Pacific Rim: Uprising",
      "genre_ids": [28, 14, 878, 12],
      "backdrop_path": "/jj8qgyrfQ12ZLZSY1PEbA3FRkfY.jpg",
      "adult": false,
      "overview":
        "It has been ten years since The Battle of the Breach and the oceans are still, but restless. Vindicated by the victory at the Breach, the Jaeger program has evolved into the most powerful global defense force in human history. The PPDC now calls upon the best and brightest to rise up and become the next generation of heroes when the Kaiju threat returns.",
      "release_date": "2018-03-21"
    },
    {
      "vote_count": 677,
      "id": 353616,
      "video": false,
      "vote_average": 6.4,
      "title": "Pitch Perfect 3",
      "popularity": 86.470489,
      "poster_path": "/fchHLsLjFvzAFSQykiMwdF1051.jpg",
      "original_language": "en",
      "original_title": "Pitch Perfect 3",
      "genre_ids": [35, 10402],
      "backdrop_path": "/f5a9YRvJ7uSM0JAtEI59sqKRnVg.jpg",
      "adult": false,
      "overview":
        "After the highs of winning the world championships, the Bellas find themselves split apart and discovering there aren't job prospects for making music with your mouth. But when they get the chance to reunite for an overseas USO tour, this group of awesome nerds will come together to make some music, and some questionable decisions, one last time.",
      "release_date": "2017-12-21"
    },
    {
      "vote_count": 1595,
      "id": 316029,
      "video": false,
      "vote_average": 8,
      "title": "The Greatest Showman",
      "popularity": 80.038653,
      "poster_path": "/dfhztJRiElqmYW4kpvjYe1gENsD.jpg",
      "original_language": "en",
      "original_title": "The Greatest Showman",
      "genre_ids": [18, 10402],
      "backdrop_path": "/zpq404Sk7qQ7N4x3xOeNgp74GtU.jpg",
      "adult": false,
      "overview":
        "The story of American showman P.T. Barnum, founder of the circus that became the famous traveling Ringling Bros. and Barnum & Bailey Circus.",
      "release_date": "2017-12-20"
    },
    {
      "vote_count": 2301,
      "id": 359940,
      "video": false,
      "vote_average": 8.2,
      "title": "Three Billboards Outside Ebbing, Missouri",
      "popularity": 57.072976,
      "poster_path": "/vgvw6w1CtcFkuXXn004S5wQsHRl.jpg",
      "original_language": "en",
      "original_title": "Three Billboards Outside Ebbing, Missouri",
      "genre_ids": [80, 18],
      "backdrop_path": "/9oRdMkQ6ysUuF7Ufb1oMH5NMDY2.jpg",
      "adult": false,
      "overview":
        "After seven months have passed without a culprit in her daughter's murder case, Mildred Hayes makes a bold move, painting three signs leading into her town with a controversial message directed at Bill Willoughby, the town's revered chief of police. When his second-in-command Officer Jason Dixon, an immature mother's boy with a penchant for violence, gets involved, the battle between Mildred and Ebbing's law enforcement is only exacerbated.",
      "release_date": "2017-11-10"
    },
    {
      "vote_count": 686,
      "id": 301337,
      "video": false,
      "vote_average": 5.1,
      "title": "Downsizing",
      "popularity": 55.656329,
      "poster_path": "/uLlmtN33rMuimRq6bu0OoNzCGGs.jpg",
      "original_language": "en",
      "original_title": "Downsizing",
      "genre_ids": [35, 18, 878],
      "backdrop_path": "/5RzW9i3vFsXCPLqYwgNJV1QuSgk.jpg",
      "adult": false,
      "overview":
        "A kindly occupational therapist undergoes a new procedure to be shrunken to four inches tall so that he and his wife can help save the planet and afford a nice lifestyle at the same time.",
      "release_date": "2017-12-22"
    },
    {
      "vote_count": 1499,
      "id": 406997,
      "video": false,
      "vote_average": 8.1,
      "title": "Wonder",
      "popularity": 47.851897,
      "poster_path": "/ouYgAatYH4JzIThj6FI3UYf31RI.jpg",
      "original_language": "en",
      "original_title": "Wonder",
      "genre_ids": [18, 10751],
      "backdrop_path": "/4rsrxYDfIzvtAsIE9wevxPRXAk4.jpg",
      "adult": false,
      "overview":
        "The story of August Pullman â€“ a boy with facial differences â€“ who enters fifth grade, attending a mainstream elementary school for the first time.",
      "release_date": "2017-11-13"
    },
    {
      "vote_count": 28,
      "id": 419831,
      "video": false,
      "vote_average": 6.2,
      "title": "I Kill Giants",
      "popularity": 45.682399,
      "poster_path": "/zNrBAF5YvzDACMHvxwHpsiAOt2I.jpg",
      "original_language": "en",
      "original_title": "I Kill Giants",
      "genre_ids": [53, 18, 14],
      "backdrop_path": "/kDcNirNDYyu2ugdVc97y5QSl7N6.jpg",
      "adult": false,
      "overview":
        "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
      "release_date": "2018-01-21"
    },
    {
      "vote_count": 1057,
      "id": 336843,
      "video": false,
      "vote_average": 7.3,
      "title": "Maze Runner: The Death Cure",
      "popularity": 43.492651,
      "poster_path": "/7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
      "original_language": "en",
      "original_title": "Maze Runner: The Death Cure",
      "genre_ids": [28, 9648, 878, 53, 12, 14],
      "backdrop_path": "/bvbyidkMaBls1LTaIWYY6UmYTaL.jpg",
      "adult": false,
      "overview":
        "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
      "release_date": "2018-01-17"
    },
    {
      "vote_count": 3,
      "id": 437543,
      "video": false,
      "vote_average": 3,
      "title": "The Monkey King 3: Kingdom of Women",
      "popularity": 41.541929,
      "poster_path": "/zMYnxPIkUVaRIuTpS77B8hLGkBB.jpg",
      "original_language": "zh",
      "original_title": "è¥¿æ¸¸è®°Â·å¥³å„¿å›½",
      "genre_ids": [28, 12, 14],
      "backdrop_path": "/xCF6Vkq1L5tTGENb07vzlQyXwKv.jpg",
      "adult": false,
      "overview":
        "The third installment of the blockbuster fantasy series sees the return of the Monkey King (Aaron Kwok) in his most action-packed adventure yet! While continuing their epic journey to the West, the Monkey King and his companions are taken captive by the Queen of an all-female land, who believes them to be part of an ancient prophecy heralding the fall of her kingdom. With a lot of sorcery and a little bit of charm, the travelers devise a plan to escape. But when their trickery angers the mighty River God, they realize they might just bring about the foretold destruction - unless they can find a way to quell her wrath.",
      "release_date": "2018-02-16"
    },
    {
      "vote_count": 461,
      "id": 401981,
      "video": false,
      "vote_average": 6.3,
      "title": "Red Sparrow",
      "popularity": 40.137189,
      "poster_path": "/fyufzd2S1lLxMxkTHTnmr6VJfe5.jpg",
      "original_language": "en",
      "original_title": "Red Sparrow",
      "genre_ids": [9648, 53],
      "backdrop_path": "/sGzuQYSTwJvLBc2PnuSVLHhuFeh.jpg",
      "adult": false,
      "overview":
        "Prima ballerina Dominika Egorova faces a bleak and uncertain future after she suffers an injury that ends her career. She soon turns to Sparrow School, a secret intelligence service that trains exceptional young people to use their minds and bodies as weapons. Egorova emerges as the most dangerous Sparrow after completing the sadistic training process. As she comes to terms with her new abilities, Dominika meets a CIA agent who tries to convince her that he is the only person she can trust.",
      "release_date": "2018-03-01"
    },
    {
      "vote_count": 1769,
      "id": 398818,
      "video": false,
      "vote_average": 8.3,
      "title": "Call Me by Your Name",
      "popularity": 39.916319,
      "poster_path": "/nPTjj6ZfBXXBwOhd7iUy6tyuKWt.jpg",
      "original_language": "en",
      "original_title": "Call Me by Your Name",
      "genre_ids": [10749, 18],
      "backdrop_path": "/vABQeCuuagIjir0YBAfL0V2XePn.jpg",
      "adult": false,
      "overview":
        "Elio Perlman is spending the summer with his family at their vacation home in Lombardy, Italy. When his father hires a handsome doctoral student, the curious 17-year-old finds himself developing a growing attraction to the young man.",
      "release_date": "2017-10-27"
    },
    {
      "vote_count": 7007,
      "id": 8587,
      "video": false,
      "vote_average": 8.1,
      "title": "The Lion King",
      "popularity": 37.014026,
      "poster_path": "/bKPtXn9n4M4s8vvZrbw40mYsefB.jpg",
      "original_language": "en",
      "original_title": "The Lion King",
      "genre_ids": [10751, 16, 18],
      "backdrop_path": "/klI0K4oQMsLhHdjA9Uw8WLugk9v.jpg",
      "adult": false,
      "overview":
        "A young lion cub named Simba can't wait to be king. But his uncle craves the title for himself and will stop at nothing to get it.",
      "release_date": "1994-06-23"
    }
  ],
  "page": 1,
  "total_results": 893,
  "dates": { "maximum": "2018-03-25", "minimum": "2018-02-06" },
  "total_pages": 45
}
