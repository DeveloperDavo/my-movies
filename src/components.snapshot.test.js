import React from "react";
import { Header, Attribution, MovieGrid, MainComponent } from "./components";
import renderer from "react-test-renderer";
import { results } from "./movies.json";
import { changeMinRating } from "./actions";
import { Provider } from "react-redux";
import { store } from "./store";
import { connect } from "react-redux";
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const slicedResults = results.slice(0, 3);

describe("Header", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Header />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("Attribution", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Attribution />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("MovieGrid", () => {
  it("should match empty snapshot", () => {
    const wrapper = shallow(<MovieGrid minRating={0} movies={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match snapshot with movies and minRating", () => {
    const wrapper = shallow(<MovieGrid minRating={7} movies={slicedResults} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("Main component", () => {
  it("should match empty snapshot", () => {
    const mainComp = (
      <MainComponent
        handleMinRatingChange={() => {}}
        handleFetchMovies={() => {}}
        minRating={6}
        movies={[]}
      />
    );
    const wrapper = shallow(mainComp);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot with movie data", () => {
    const mainComp = (
      <MainComponent
        handleMinRatingChange={() => {}}
        handleFetchMovies={() => {}}
        minRating={8}
        movies={slicedResults}
      />
    );
    const wrapper = shallow(mainComp);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
