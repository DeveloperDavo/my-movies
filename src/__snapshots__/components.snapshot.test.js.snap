// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribution should match snapshot 1`] = `
<styled.footer>
  <styled.img
    alt="tmdb logo"
    src="tmdb-logo.png"
  />
  <styled.div>
    This product uses the TMDb API but is not endorsed or certified by TMDb.
  </styled.div>
</styled.footer>
`;

exports[`Header should match snapshot 1`] = `
<styled.header>
  <styled.h1>
    My Movies
  </styled.h1>
</styled.header>
`;

exports[`Main component should match empty snapshot 1`] = `
<main>
  <styled.div>
    Movies now playing
  </styled.div>
  <MinRatingSelect
    onChange={[Function]}
    value={6}
  />
  <Component
    minRating={6}
    movies={Array []}
  />
  <Component />
</main>
`;

exports[`Main component should match snapshot with movie data 1`] = `
<main>
  <styled.div>
    Movies now playing
  </styled.div>
  <MinRatingSelect
    onChange={[Function]}
    value={8}
  />
  <Component
    minRating={8}
    movies={
      Array [
        Object {
          "adult": false,
          "backdrop_path": "/9ywA15OAiwjSTvg3cBs9B7kOCBF.jpg",
          "genre_ids": Array [
            18,
            10749,
          ],
          "id": 337167,
          "original_language": "en",
          "original_title": "Fifty Shades Freed",
          "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
          "popularity": 572.203541,
          "poster_path": "/jjPJ4s3DWZZvI4vw8Xfi4Vqa1Q8.jpg",
          "release_date": "2018-02-07",
          "title": "Fifty Shades Freed",
          "video": false,
          "vote_average": 6.1,
          "vote_count": 1097,
        },
        Object {
          "adult": false,
          "backdrop_path": "/b6ZJZHUdMEFECvGiDpJjlfUWela.jpg",
          "genre_ids": Array [
            28,
            12,
            14,
            878,
          ],
          "id": 284054,
          "original_language": "en",
          "original_title": "Black Panther",
          "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
          "popularity": 329.754985,
          "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
          "release_date": "2018-02-13",
          "title": "Black Panther",
          "video": false,
          "vote_average": 7.4,
          "vote_count": 3477,
        },
        Object {
          "adult": false,
          "backdrop_path": "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
          "genre_ids": Array [
            12,
            35,
            10751,
            16,
          ],
          "id": 354912,
          "original_language": "en",
          "original_title": "Coco",
          "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
          "popularity": 211.906826,
          "poster_path": "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          "release_date": "2017-10-27",
          "title": "Coco",
          "video": false,
          "vote_average": 7.8,
          "vote_count": 3404,
        },
      ]
    }
  />
  <Component />
</main>
`;

exports[`MovieGrid should match empty snapshot 1`] = `<styled.div />`;

exports[`MovieGrid should match snapshot with movies and minRating 1`] = `
<styled.div>
  <Component
    imageSrc="http://image.tmdb.org/t/p/w200//uxzzxijgPIY7slzFvMotPv8wjKA.jpg"
    key="284054"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/b6ZJZHUdMEFECvGiDpJjlfUWela.jpg",
        "genre_ids": Array [
          28,
          12,
          14,
          878,
        ],
        "id": 284054,
        "original_language": "en",
        "original_title": "Black Panther",
        "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
        "popularity": 329.754985,
        "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
        "release_date": "2018-02-13",
        "title": "Black Panther",
        "video": false,
        "vote_average": 7.4,
        "vote_count": 3477,
      }
    }
  />
  <Component
    imageSrc="http://image.tmdb.org/t/p/w200//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
    key="354912"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
        "genre_ids": Array [
          12,
          35,
          10751,
          16,
        ],
        "id": 354912,
        "original_language": "en",
        "original_title": "Coco",
        "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
        "popularity": 211.906826,
        "poster_path": "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
        "release_date": "2017-10-27",
        "title": "Coco",
        "video": false,
        "vote_average": 7.8,
        "vote_count": 3404,
      }
    }
  />
</styled.div>
`;
